server/src/main/java/org/elasticsearch/search/fetch/subphase/FieldFetcher.java

```diff
@@ -45,6 +45,7 @@ public static FieldFetcher create(SearchExecutionContext context, Collection<Fie
         List<String> unmappedFetchPattern = new ArrayList<>();
         List<ResolvedField> resolvedFields = new ArrayList<>();
 
+        context.setAllowedFields(context::isMetadataField);
```

Comment:
I had to add this here because otherwise some tests fail because a NPE. The NPE happens later on when in FIeldFetcher we do

```
MappedFieldType ft = context.getFieldType(field);
```

this would return null for metadata fields and will result in an NPE when calling


```
return fieldAndFormat.ft.valueFetcher(context, fieldAndFormat.format);
```
in buildValueFetcher.

Reply:
The explanation I have for this is that for metadata fields which are returned by default we need them to be "allowed". Honestly it is not really clear to me what "being allowed" means in this context...I can only see the flag `allowedFieields` (predicate) is used in the `QueryRewriteContext#fieldType` method which I interpret that "getting the type for a field might be allowed or not"...but I don't know the reason why.

Reply:
I reverted the change because it was breaking more tests then it was fixing.